This is a patch for the following issue:
http://trac.calendarserver.org/ticket/416

It was caused by somehow incorrect lineending handling.

Read from my IMAP, I have message containing \r\n when they get read by message_from_string() 
They are delimited by \n in the header and still \r\n in the content. 

Sending the mail via ESMTPSenderFactory() now creates two \n into \r\n as of RFC2822, but does 
so for the content too, therefore too many line breaks are in there.

One MUST NOT use str(msg) as this contains the unix header (envelope) and therefore the headers
will be added twice ...


Index: twistedcaldav/mail.py
===================================================================
--- twistedcaldav/mail.py	(Revision 7046)
+++ twistedcaldav/mail.py	(Arbeitskopie)
@@ -742,15 +742,12 @@
                 contextFactory = None
 
             deferred = defer.Deferred()
-            del msg["From"]
-            msg["From"] = fromAddr
+            msg.replace_header("From", fromAddr)
+            msg.replace_header("To", toAddr)
             del msg["Reply-To"]
-            msg["Reply-To"] = fromAddr
-            del msg["To"]
-            msg["To"] = toAddr
             factory = ESMTPSenderFactory(
                 settings["Username"], settings["Password"],
-                fromAddr, toAddr, StringIO(str(msg)), deferred,
+                fromAddr, toAddr, StringIO(msg.as_string().replace("\r\n","\n")), deferred,
                 contextFactory=contextFactory,
                 requireAuthentication=False,
                 requireTransportSecurity=settings["UseSSL"],
